Sources/Common/Core/Math/index.js

```diff
@@ -577,47 +564,57 @@ export function multiplyMatrix(a, b, rowA, colA, rowB, colB, out_rowXcol) {
     vtkErrorMacro('Number of columns of A must match number of rows of B.');
   }
 
+  const copyA = createArray(9);
+  const copyB = createArray(9);
+  for (let j = 0; j < rowA * colA; j++) {
+    copyA[j] = a[j];
+  }
+  for (let j = 0; j < rowB * colB; j++) {
+    copyB[j] = b[j];
+  }
   // output matrix is rowA*colB
   // output row
   for (let i = 0; i < rowA; i++) {
     // output col
     for (let j = 0; j < colB; j++) {
-      out_rowXcol[i][j] = 0;
+      out_rowXcol[i * colB + j] = 0;
       // sum for this point
       for (let k = 0; k < colA; k++) {
-        out_rowXcol[i][j] += a[i][k] * b[k][j];
+        out_rowXcol[i * colB + j] += copyA[i * colA + k] * copyB[j + colB * k];
       }
     }
   }
 }
 
 export function transpose3x3(in_3x3, outT_3x3) {
```

Comment:
why not keep the same logic as the previous implementation (using a tmp value)?

Reply:
Do you mean avoiding the copy of a and b by storing the result in a temporary Array and then copy it into out_rowXcol ?